### ISSUE TITLE
`max-len`: This rule does not work for array or object when ignoreStrings:true is configured.

### ISSUE THREAD

**Issue by pikadun** [eslint-stylistic/eslint-stylistic#515]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

```js
/* eslint @stylistic/max-len: ["error", { "code": 80 ,ignoreStrings: true}] */

const foo = { bar: "This is a bar.", baz: { qux: "This is a qux" }, difficult: "to read" };
const arr = ["This is a bar.", "This is a qux", "to read", "This is a bar.", "This is a qux", "to read"];
```

### Reproduction

https://stackblitz.com/edit/github-krrscl?file=src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [X] I am willing to submit a PR with failing tests

**Comment by evgencode** [comment]
Also, auto-fixing of long lines does not work if you run `eslint --fix` or via IDE
By analogy with the prettier.

code example:
max-length: 100
```
if (Object.keys(prevArgs).length === Object.keys(props).length && Object.keys(prevArgs).some((key) => !Object.is(prevArgs[key], props[key]))) {
            res = fn(props);
}
```

**Comment by CristianBonilla** [comment]
I'm also experiencing the same problem, not only with arrays and objects, but with all code in general...

And running `eslint --fix` doesn't work

Here are some examples:

```typescript
// 1.
import { createLogger, format, transports, type LoggerOptions, type Logger as WinstonLogger } from 'winston';

// 2.
const getSplat = (info: TransformableInfo) => info[Symbol.for('splat')] as [LoggerSpecificOptionsWithType, ...unknown[]];

// 3.
export type RouteHandler = (instance: FastifyInstance, options: RouteShorthandOptions & Record<never, never>, handler: (error?: Error) => void) => void;
```

**Comment by 9romise** [comment]
About auto-fix: #73
